aws-ts-serverless-raw/index.ts

```diff
@@ -27,31 +27,44 @@ let counterTable = new aws.dynamodb.Table("counterTable", {
 // Lambda Function
 ///////////////////
 
-// Create a Role giving our Lambda access.
-let policy: aws.iam.PolicyDocument = {
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Action": "sts:AssumeRole",
-            "Principal": {
-                "Service": "lambda.amazonaws.com",
-            },
-            "Effect": "Allow",
-            "Sid": "",
-        },
-    ],
-};
-let role = new aws.iam.Role("mylambda-role", {
-    assumeRolePolicy: JSON.stringify(policy),
+// Give our Lambda access to the Dynamo DB table, CloudWatch Logs and Metrics.
+const role = new aws.iam.Role("mylambda-role", {
+    assumeRolePolicy: pulumi.output(
+        aws.iam.getPolicyDocument({
+            statements: [{
+                actions: ["sts:AssumeRole"],
+                principals: [{
+                    identifiers: ["lambda.amazonaws.com"], 
+                    type: "Service",
+                }],
+                effect: "Allow",
+            }],
+        })
+    ).apply(r => r.json),
 });
-let fullAccess = new aws.iam.RolePolicyAttachment("mylambda-access", {
+
+const policy = new aws.iam.Policy("mylambda-policy", {
+    policy: counterTable.arn.apply(arn => aws.iam.getPolicyDocument({
```

Comment:
You should not need `getPolicyDocument`. That is inherited from the Terraform provider, but should be unneeded in Pulumi since you can write JSON more directly. 

In fact this whole thing could be:

`policy: pulumi.output({ ...})`

Where the object inside can embed `counterTable.arm`.

Reply:
`policy` is of type `string`, so somebody has to check types first and then convert it?

Reply:
Ahh - that sounds like a bug - opened https://github.com/pulumi/pulumi-aws/issues/583.

But the reason I thought it did was actually that this could (and is in most of our other examples) be simplified to `new aws.iam.RolePolicy()` instead of `aws.iam.Policy + aws.iam.RolePolicyAttachment`.  The former is both shorter and does already support passing JSON directly.